src/lib/moment/get-set.js

```diff
@@ -27,11 +29,27 @@ export function set (mom, unit, value) {
 
 // MOMENTS
 
-export function getSet (units, value) {
-    var unit;
+export function stringGet (units) {
+    units = normalizeUnits(units);
+    if (isFunction(this[units])) {
+        return this[units]();
+    }
+    return this;
+}
+
+
+export function stringSet (units, value) {
     if (typeof units === 'object') {
-        for (unit in units) {
-            this.set(unit, units[unit]);
+        units = normalizeObjectUnits(units);
+        // for (unit in units) {
+        //     this.set(unit, units[unit]);
+        // }
+        var prioritized = getPrioritizedUnits();
+        for (var i = 0; i < prioritized.length; i++)
```

Comment:
I'm not sure you need a callback, but I like the idea of sorting the units provided instead of the units known. For each unit, look it up in the map that gives you the priority; that gives you a list of unit/priority pairs. Sort the list by priority. IOW, `getPrioritizedUnit(units)` returns just those units in the right order.